sqldb/sqlc/queries/invoices.sql

```diff
@@ -26,7 +26,11 @@ WHERE invoice_id = $1;
 -- name: GetInvoice :many
 SELECT i.*
 FROM invoices i
-LEFT JOIN amp_sub_invoices a on i.id = a.invoice_id
+LEFT JOIN amp_sub_invoices a 
+ON i.id = a.invoice_id
+AND (
+    a.set_id = sqlc.narg('set_id') OR sqlc.narg('set_id') IS NULL
```

Comment:
Why wasn't the filter in the `WHERE` clause enough?

Reply:
I think it's the easiest to explain with a fiddle: https://www.db-fiddle.com/f/oLWkxkjhRRXirjXD7jicqf/0
Essentially we wanted to capture the case when the invoice is there but it doesn't yet have any sub invoices. In this case `UpdateInvoice` will attempt to fetch the invoice but with the old query it'd fail to do so.